### PR TITLE
fix(warning): remove warnings in test

### PR DESCRIPTION
**Pull request by trollepierre**
- Adds ConsoleUsageReporter that verify console is not used in test
- Quick fixes easy test to prevent spying console for real errors
- Spying console only when tests could not be easily fixed
- throw Node unhandledRejection errors during test to detect them

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -21,7 +22,7 @@ module.exports = {
     '^folder-references(.*)': '<rootDir>/src/folder-references$1',
     '^lib(.*)': '<rootDir>/src/lib$1',
     'react-pdf/dist/pdf.worker.entry.js':
-    '<rootDir>/jestHelpers/mocks/pdfjsWorkerMock.js',
+      '<rootDir>/jestHelpers/mocks/pdfjsWorkerMock.js',
```
Comment (Crash--) [inline jest.config.js:25]: tab should be removed?

File: src/components/Error/Empty.jsx [hunk 2]
```diff
@@ -25,6 +25,8 @@ const EmptyCanvas = ({ type, canUpload, localeKey }) => {
         (localeKey && t(`empty.${localeKey}_text`)) ||
         (canUpload && t('empty.text'))
       }
+      layout={false} //  Please add layout=false in the props of the Empty component, and implement the vertical
```
Comment (Crash--) [inline src/components/Error/Empty.jsx:28]: I think we can't just do that. We need t implement the vertical centering, no?

File: src/drive/lib/migration/qualification.js [hunk 3]
```diff
@@ -100,15 +100,15 @@ export const extractFilesToMigrate = files => {
...
+ * @param {string} oldLabel - The old qualification label
  * @returns {string} The new qualification label
  */
 const getNewLabelSetFromCozyScanner = oldLabel => {
   if (oldLabel === 'registration') {
     return 'vehicle_registration'
   }
   if (oldLabel === 'insurance_card') {
-    return 'national_insurance_card'
+    return 'national_health_insurance_card'
```
Comment (Crash--) [inline src/drive/lib/migration/qualification.js:111]: If a PR depends on another one, can you change the target branch for this one? In order to not review the same thing. Once the previous will be merged, the target branche will be automatically set to master here.

File: src/drive/web/modules/move/MoveModal.spec.jsx [hunk 4]
```diff
@@ -135,15 +139,19 @@ describe('MoveModal component', () => {
...
+      try {
+        const callback = jest.fn()
+        const component = setupComponent()
+        await component
+          .instance()
+          .cancelMove([], ['trashed-1', 'trashed-2'], callback)
+        // expect(collectionSpy).toHaveBeenCalledWith('io.cozy.files')
+        // expect(restoreSpy).toHaveBeenCalledWith('trashed-1')
+        // expect(restoreSpy).toHaveBeenCalledWith('trashed-2')
+        // expect(callback).toHaveBeenCalled()
```
Comment (Crash--) [inline src/drive/web/modules/move/MoveModal.spec.jsx:151]: should be restored?

File: src/photos/targets/public/App.spec.jsx [hunk 5]
```diff
@@ -1,11 +1,14 @@
...
 import { shallow } from 'enzyme'
 import { App } from './App'
+import { createMockClient } from 'cozy-client'
 
 describe('Public view', () => {
+  const client = createMockClient({})
   const options = {
     context: {
-      t: jest.fn()
+      t: jest.fn().mockReturnValue('some translation'),
```
Comment (Crash--) [inline src/photos/targets/public/App.spec.jsx:10]: generally we go for: `t = x => x`

File: jestHelpers/ConsoleUsageReporter.js [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+      } else {
+        throw error
+      }
+    }
+  }
+
+  onRunStart() {
+    ConsoleUsageReporter.deleteTemporaryFile()
+  }
+}
```
Comment (Crash--) [inline jestHelpers/ConsoleUsageReporter.js:172]: I don't understand what this file really do 🤷‍♂️

File: jest.config.js (hunk 1 above)
Comment (trollepierre) [reply jest.config.js:25]: this line has been automatically prettied when I commited the file, apparently 🤔 . I have no opinion on it.

File: jestHelpers/ConsoleUsageReporter.js (hunk 6 above)
Comment (trollepierre) [reply jestHelpers/ConsoleUsageReporter.js:172]: Whenever a console.error is thrown while running jest, the ConsoleUsageReporter will intercept it and make the test failing and indicating how to solve the issue by mocking the console for example.

The advantage of this, is to fight against breaking changes. Whenever a breaking changes occur while upgrading dependencies, the warning in the console - that indicates the breaking changes - will make the "unit test" jobs failing.

File: src/components/Error/Empty.jsx (hunk 2 above)
Comment (trollepierre) [reply src/components/Error/Empty.jsx:28]: I succeed in implementing quickly the fix. Thanks

File: jestHelpers/ConsoleUsageReporter.js (hunk 6 above)
Comment (trollepierre) [reply jestHelpers/ConsoleUsageReporter.js]: I remove the file. This PR can be merged, I guess

**Review by Crash-- — APPROVED** [review]

**Pull request merged**